### PR TITLE
Fix some style issues given by clippy

### PR DESCRIPTION
**Pull request by neosilky**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -384,7 +384,7 @@ impl Config {
 
         let mut objects = Vec::new();
         let mut src_dst = Vec::new();
-        for file in self.files.iter() {
```
Comment (alexcrichton) [inline src/lib.rs]: I personally prefer `for x in foo.iter()` over `for x in &foo`, can this remain as it was?

File: src/lib.rs [hunk 2]
```diff
@@ -963,9 +962,7 @@ impl Config {
             let target = self.get_target();
             if target.contains("msvc") {
                 None
-            } else if target.contains("darwin") {
-                Some("c++".to_string())
```
Comment (alexcrichton) [inline src/lib.rs]: Can these remain separate?

**Comment by neosilky** [comment]
@alexcrichton Sorry for the late reply. I have removed those commits and force pushed. Should be good to go! :+1:

**Comment by alexcrichton** [comment]
Looks like there's some errors on CI?

**Comment by neosilky** [comment]
@alexcrichton Sorry about the super late reply... I forgot I had nightly as the default. Should be good to go now :+1:

**Comment by alexcrichton** [comment]
Thanks!

**Pull request merged**